scrnaseq.Rmd

```diff
@@ -688,45 +721,62 @@ if (param$cc_remove) {
-sc = purrr::map(sc, function(s) { 
-  Seurat::ScaleData(s, features=ifelse(param$integrate_samples[["method"]]=="single", rownames(s[["RNA"]]), VariableFeatures(s)), vars.to.regress=param$vars_to_regress, verbose=FALSE) 
+```{r part3_normalisation, results="hide", warning=FALSE}
+if (param$norm == "RNA") { 
+  # Find variable features from normalised data (unaffected by scaling)
+  sc = purrr::map(sc, Seurat::FindVariableFeatures, selection.method = "vst", nfeatures = 3000, verbose=FALSE)
+  
+  # Scale 
+  # Note: For a single dataset where no integration is needed, all features can already be scaled here. 
+  #   Otherwise, it is enough to scale variable features only and scaling of all features will be done later.
+  sc = purrr::map(sc, function(s) { 
+    Seurat::ScaleData(s, 
+                      features=ifelse(param$integrate_samples[["method"]]=="single", rownames(s[["RNA"]]), VariableFeatures(s)), 
+                      vars.to.regress=param$vars_to_regress, 
+                      verbose=FALSE) 
   })
-
-# Run SCTransform
-#
-# This is a new normalisation method that replaces previous Seurat functions 'NormalizeData', 'FindVariableFeatures', and 'ScaleData'. 
-# vignette: https://satijalab.org/seurat/v3.0/sctransform_vignette.html
-# paper: https://www.biorxiv.org/content/10.1101/576827v2
-# Normalised data end up here: sc@assays$SCT@data
-# Note: For a single dataset where no integration is needed, all features can already be scaled here. Otherwise, it is enough to scale only the variable features.
-sc = purrr::map(list_names(sc), function(n) { SCTransform(sc[[n]], vars.to.regress=param$vars_to_regress, min_cells=param$feature_filter[[n]][["min_cells"]], verbose=FALSE, return.only.var.genes=!(param$integrate_samples[["method"]]=="single")) })
-
-# Note: It is not guaranteed that all genes are successfully normalised with SCTransform. Consequently, some genes might be missing from the SCT assay. See: https://github.com/ChristophH/sctransform/issues/27
+  
+} else if (param$norm == "SCT") {
+  # Run SCTransform
+  #
+  # This is a new normalisation method that replaces previous Seurat functions 'NormalizeData', 'FindVariableFeatures', and 'ScaleData'. 
+  # vignette: https://satijalab.org/seurat/v3.0/sctransform_vignette.html
+  # paper: https://www.biorxiv.org/content/10.1101/576827v2
+  # Normalised data end up here: sc@assays$SCT@data
+  # Note: For a single dataset where no integration is needed, all features can already be scaled here. 
+  #   Otherwise, it is enough to scale only the variable features.
+  # Note: It is not guaranteed that all genes are successfully normalised with SCTransform. 
+  #   Consequently, some genes might be missing from the SCT assay. 
+  #   See: https://github.com/ChristophH/sctransform/issues/27
+sc = purrr::map(list_names(sc), function(n) { 
+  SCTransform(sc[[n]], 
```

Comment:
Sometimes, there will be many warnings. Is there something like a function to truncate, i.e. show only the first 5?

Reply:
At some point, we had suppressWarnings, and we removed it for this PR. Should I add it back? At least in the chunk `multi_dataset_integration_merge`, we don't set `warning=FALSE`.

Reply:
In the above mentioned chunk, I added `suppressWarnings`.